peer.go

```diff
@@ -91,6 +95,13 @@ type channelCloseUpdate struct {
 	Success     bool
 }
 
+// timestampError is a timestamped error that is used to cache the most recent
+// errors sent from our peers.
+type timestampError struct {
```

Comment:
I think I'll pass for this case since this isn't really an error that we're propagating up through a call stack. Would just need to add an `Error` and `Unwrap` function to this struct, and I think it server it purpose as is?